crates/ironrdp-pdu/src/rdp/suppress_output.rs

```diff
@@ -42,14 +45,17 @@ impl PduParsing for SuppressOutputPdu {
         } else {
             None
         };
-        Ok(Self {
-            allow_display_updates,
-            desktop_rect,
-        })
+        Ok(Self { desktop_rect })
     }
 
     fn to_buffer(&self, mut stream: impl io::Write) -> Result<(), Self::Error> {
-        stream.write_u8(self.allow_display_updates.to_u8().unwrap())?;
+        let allow_display_updates = if self.desktop_rect.is_some() {
+            AllowDisplayUpdatesType::AllowDisplayUpdates
+        } else {
+            AllowDisplayUpdatesType::SuppressDisplayUpdates
+        };
+
+        stream.write_u8(allow_display_updates.as_u8())?;
```

Comment:
Note: eliminates the invalid state `allow_display_updates = SuppressDisplayUpdates` + `desktop_rect = Some(…)`.